src/openms/include/OpenMS/DATASTRUCTURES/StringUtils.h

```diff
@@ -124,7 +124,7 @@ namespace OpenMS
     {
       String res;
       size_t count = 0;
-      while (count < length && *(s + count) != 0)
```

Comment:
@cbielow @timosachsenberg I hope you agree with this change, it aligns us with `std::string`